patches/chromium/feat_add_streaming-protocol_registry_to_multibuffer_data_source.patch

```diff
@@ -0,0 +1,97 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Paul Frazee <pfrazee@gmail.com>
+Date: Thu, 7 May 2020 11:15:51 -0500
+Subject: feat: add streaming-protocol registry to multibuffer_data_source
+
+blink::WebMediaPlayerImpl - which provides the <video> and <audio> behaviors - needs to know
+whether a data source will stream or fully buffer the response. It determines this behavior
+with MultibufferDataSource::AssumeFullyBuffered() which has http/s hardwired. An incorrect
+determination will cause the video/audio to fail playing.
+
+This patch adds a list of "streaming protocols" to the MultibufferDataSource in order to allow
+other protocols to register their streaming behavior. MultibufferDataSource::AssumeFullyBuffered()
+then refers to the list so that it can correctly determine the data source's settings.
+
+diff --git a/media/blink/multibuffer_data_source.cc b/media/blink/multibuffer_data_source.cc
+index 0f6ae1fb8b4ff9f24ce3f407b7359e016fc6de5f..fb4233f206e564057301f6a9224b7dc2aebca868 100644
+--- a/media/blink/multibuffer_data_source.cc
++++ b/media/blink/multibuffer_data_source.cc
+@@ -10,6 +10,7 @@
+ #include "base/callback_helpers.h"
+ #include "base/location.h"
+ #include "base/macros.h"
++#include "base/no_destructor.h"
+ #include "base/numerics/ranges.h"
+ #include "base/numerics/safe_conversions.h"
+ #include "base/single_thread_task_runner.h"
+@@ -65,10 +66,40 @@ const int kUpdateBufferSizeFrequency = 32;
+ // How long to we delay a seek after a read?
+ constexpr base::TimeDelta kSeekDelay = base::TimeDelta::FromMilliseconds(20);
+ 
++// List of currently registered schemes and associated properties.
++struct SchemeRegistry {
++  // Schemes that can stream their responses.
++  std::vector<std::string> streaming_schemes = {
++    url::kHttpsScheme,
++    url::kHttpScheme
++  };
++};
++
++// Ensure that the schemes aren't modified after first use.
++static std::atomic<bool> g_scheme_registries_used{false};
++
++// Gets the scheme registry without locking the schemes. This should *only* be
++// used for adding schemes to the registry.
++SchemeRegistry* GetSchemeRegistryWithoutLocking() {
++  static base::NoDestructor<SchemeRegistry> registry;
++  return registry.get();
++}
```

Comment:
Since we only have one thing in our registry, there's no need for a struct. Instead:
```suggestion
+std::vector<std::string>* GetStreamingSchemes() {
+  static base::NoDestructor<std::vector<std::string>> streaming_schemes;
+  return streaming_schemes.get();
+}
```

Reply:
Done, I had to pre-populate the list with http/s, lmk if my solution works for yall